core/src/test/java/org/elasticsearch/action/get/GetResponseTests.java

```diff
@@ -0,0 +1,357 @@
+            fields.put(getField.getName(), getField);
+            expectedFields.put(expectedGetField.getName(), expectedGetField);
+        }
+        return Tuple.tuple(fields, expectedFields);
+    }
+
+    private static Tuple<GetField, GetField> randomGetField() {
+        if (randomBoolean()) {
+            String fieldName = randomFrom(ParentFieldMapper.NAME, RoutingFieldMapper.NAME, UidFieldMapper.NAME);
+            GetField getField = new GetField(fieldName, Collections.singletonList(randomAsciiOfLengthBetween(3, 10)));
+            return Tuple.tuple(getField, getField);
+        }
+        String fieldName = randomAsciiOfLengthBetween(3, 10);
+        int numValues = randomIntBetween(1, 5);
+        List<Object> values = new ArrayList<>();
+        List<Object> expectedValues = new ArrayList<>();
+        int dataType = randomIntBetween(0, 8);
+        for (int i = 0; i < numValues; i++) {
+            switch(dataType) {
+                case 0:
+                    long randomLong = randomLong();
+                    values.add(randomLong);
+                    expectedValues.add(randomLong);
+                    break;
+                case 1:
+                    int randomInt = randomInt();
+                    values.add(randomInt);
+                    expectedValues.add(randomInt);
+                    break;
+                case 2:
+                    Short randomShort = randomShort();
+                    values.add(randomShort);
+                    expectedValues.add(randomShort.intValue());
+                    break;
+                case 3:
+                    Byte randomByte = randomByte();
+                    values.add(randomByte);
+                    expectedValues.add(randomByte.intValue());
+                    break;
+                case 4:
+                    double randomDouble = randomDouble();
+                    values.add(randomDouble);
+                    expectedValues.add(randomDouble);
+                    break;
+                case 5:
+                    boolean randomBoolean = randomBoolean();
+                    values.add(randomBoolean);
+                    expectedValues.add(randomBoolean);
+                    break;
+                case 6:
+                    Date randomDate = new Date(randomPositiveLong());
+                    values.add(randomDate);
+                    expectedValues.add(DEFAULT_DATE_PRINTER.print(randomDate.getTime()));
+                    break;
+                case 7:
+                    String randomString = randomUnicodeOfLengthBetween(3, 10);
+                    values.add(randomString);
+                    expectedValues.add(randomString);
+                    break;
+                case 8:
+                    String randomText = randomUnicodeOfLengthBetween(3, 10);
+                    values.add(new Text(randomText));
+                    expectedValues.add(randomText);
+                    break;
```

Comment:
I think we need to add a value type here that doesn't render to a simple json value but to an object, e.g. GeoPoint.